pkg/apis/admissionregistration/validation/validation.go

```diff
@@ -1151,9 +1151,34 @@ func validateParamRef(pr *admissionregistration.ParamRef, fldPath *field.Path) f
 	if pr == nil {
 		return allErrors
 	}
-	for _, msg := range path.ValidatePathSegmentName(pr.Name, false) {
-		allErrors = append(allErrors, field.Invalid(fldPath.Child("name"), pr.Name, msg))
+
+	if len(pr.Name) > 0 {
+		for _, msg := range path.ValidatePathSegmentName(pr.Name, false) {
+			allErrors = append(allErrors, field.Invalid(fldPath.Child("name"), pr.Name, msg))
+		}
+
+		if pr.Selector != nil {
+			allErrors = append(allErrors, field.Invalid(fldPath.Child("name"), pr.Name, `must be unset if "selector" is set`))
```

Comment:
I was referring to 
https://github.com/kubernetes/kubernetes/blob/9946ea9fd851fb1cd2148b746077d9f89c27e074/staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go#L886

Maybe a forbidden error with detail like `name and selector are mutual exclusive` makes more sense?

Reply:
ty! changed